tests/test_workers.py

```diff
@@ -0,0 +1,238 @@
+from __future__ import annotations
+
+import signal
+import subprocess
+import sys
+import tempfile
+import time
+from typing import TYPE_CHECKING
+
+import httpx
+import pytest
+
+if TYPE_CHECKING:
+    from ssl import SSLContext
+    from typing import IO, Generator
+
+    from asgiref.typing import (
+        ASGIReceiveCallable,
+        ASGISendCallable,
+        LifespanStartupFailedEvent,
+        Scope,
+    )
+
+pytestmark = pytest.mark.skipif(sys.platform == "win32", reason="requires unix")
+gunicorn_arbiter = pytest.importorskip("gunicorn.arbiter", reason="requires gunicorn")
+uvicorn_workers = pytest.importorskip("uvicorn.workers", reason="requires gunicorn")
```

Comment:
`pytest.importorskip()` is used so that tests are skipped if Gunicorn is not installed. Test coverage will obviously be lower if these tests don't run, but at least it makes the test suite able to run without Gunicorn.

Reply:
This is fine.